### PR TITLE
net-mgmt/lldpd: New plugin

### PR DESCRIPTION
**Pull request by mimugmail**
- Add an Enable/Disable button for the daemon itself and the protocols it supports
- Add a Diagnostics tab which shows the current neighbors.

@fichtner Some commits ago I had an issue with save button on the wrong position, it's again placed wrong .. do you know what's missing? :)

image

### CODE REVIEW

File: net-mgmt/lldpd/src/opnsense/scripts/OPNsense/Lldpd/setup.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+
+mkdir -p /var/db/collectd
```
Comment (fichtner) [inline net-mgmt/lldpd/src/opnsense/scripts/OPNsense/Lldpd/setup.sh]: collectd? :)

File: net-mgmt/lldpd/src/opnsense/service/conf/actions.d/actions_lldpd.conf [hunk 2]
```diff
@@ -0,0 +1,35 @@
+[start]
+command:/usr/local/opnsense/scripts/OPNsense/Lldpd/setup.sh;/usr/local/etc/rc.d/lldpd start
+parameters:
+type:script
+message:starting Lldpd
+
+[stop]
+command:/usr/local/etc/rc.d/lldpd stop; exit 0
```
Comment (fichtner) [inline net-mgmt/lldpd/src/opnsense/service/conf/actions.d/actions_lldpd.conf]: does this really require masking the exit code?

File: net-mgmt/lldpd/src/opnsense/service/templates/OPNsense/Lldpd/lldpd [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{% if helpers.exists('OPNsense.lldpd.general.enabled') and OPNsense.lldpd.general.enabled == '1' %}
+lldpd_opnsense_bootup_run="/usr/local/opnsense/scripts/OPNsense/Lldpd/setup.sh"
```
Comment (fichtner) [inline net-mgmt/lldpd/src/opnsense/service/templates/OPNsense/Lldpd/lldpd]: this one can be omitted, was renamed to _var_script :)

File: net-mgmt/lldpd/src/opnsense/mvc/app/views/OPNsense/Lldpd/general.volt [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+#}
+<ul class="nav nav-tabs" data-tabs="tabs" id="maintabs">
+    <li class="active"><a data-toggle="tab" href="#general">{{ lang._('General') }}</a></li>
+    <li><a data-toggle="tab" href="#neighbor">{{ lang._('Neighbors') }}</a></li>
+</ul>
+
+<div class="tab-content content-box tab-content">
+    <div id="general" class="tab-pane fade in active">
+        {{ partial("layout_partials/base_form",['fields':generalForm,'id':'frm_general_settings'])}}
+        <hr />
```
Comment (fichtner) [inline net-mgmt/lldpd/src/opnsense/mvc/app/views/OPNsense/Lldpd/general.volt]: boxing here is wrong for your horizontal line and button:

https://github.com/opnsense/plugins/blob/master/mail/postfix/src/opnsense/mvc/app/views/OPNsense/Postfix/general.volt#L49

File: net-mgmt/lldpd/src/opnsense/service/templates/OPNsense/Lldpd/lldpd.conf [hunk 5]
```diff
@@ -0,0 +1,2 @@
+{% if helpers.exists('OPNsense.lldpd.general.enabled') and OPNsense.lldpd.general.enabled == '1' %}
```
Comment (fichtner) [inline net-mgmt/lldpd/src/opnsense/service/templates/OPNsense/Lldpd/lldpd.conf]: will this ever receive content?

Comment (mimugmail) [reply net-mgmt/lldpd/src/opnsense/service/templates/OPNsense/Lldpd/lldpd.conf]: Yes, you can set Hostname etc. to manipulate the default values, but for version 1.0 it's not really required, we'll see if there's a need for it.

File: net-mgmt/lldpd/src/opnsense/service/templates/OPNsense/Lldpd/lldpd (hunk 3 above)
Comment (mimugmail) [reply net-mgmt/lldpd/src/opnsense/service/templates/OPNsense/Lldpd/lldpd]: Can you give me an example please ...

File: net-mgmt/lldpd/src/opnsense/service/conf/actions.d/actions_lldpd.conf (hunk 2 above)
Comment (mimugmail) [reply net-mgmt/lldpd/src/opnsense/service/conf/actions.d/actions_lldpd.conf]: I'll try it :)

File: net-mgmt/lldpd/src/opnsense/service/templates/OPNsense/Lldpd/lldpd (hunk 3 above)
Comment (fichtner) [reply net-mgmt/lldpd/src/opnsense/service/templates/OPNsense/Lldpd/lldpd]: just delete the line, everything else is fine

File: net-mgmt/lldpd/src/opnsense/mvc/app/views/OPNsense/Lldpd/general.volt (hunk 4 above)
Comment (mimugmail) [reply net-mgmt/lldpd/src/opnsense/mvc/app/views/OPNsense/Lldpd/general.volt]: Done, thanks

File: net-mgmt/lldpd/src/opnsense/service/templates/OPNsense/Lldpd/lldpd (hunk 3 above)
Comment (mimugmail) [reply net-mgmt/lldpd/src/opnsense/service/templates/OPNsense/Lldpd/lldpd]: OK, done

File: net-mgmt/lldpd/src/opnsense/scripts/OPNsense/Lldpd/setup.sh (hunk 1 above)
Comment (mimugmail) [reply net-mgmt/lldpd/src/opnsense/scripts/OPNsense/Lldpd/setup.sh]: The script can be ignored, I updated all the relating files and removed it.

**Comment by mimugmail** [comment]
@fichtner I added an own function in the servicecontroller:
https://github.com/mimugmail/plugins/blob/lldpd/net-mgmt/lldpd/src/opnsense/mvc/app/controllers/OPNsense/Lldpd/Api/ServiceController.php#L44-L53

Can this stay there with the new mutable design?

**Comment by fichtner** [comment]
Yes, the mutable service only provides start, stop, restart, status and reconfigure.

**Comment by mimugmail** [comment]
Hm, when I change to new mutable I get:

```
[29-Dec-2017 07:12:12 Europe/Berlin] PHP Notice:  Access to undefined property translator in /usr/local/opnsense/mvc/app/controllers/OPNsense/Base/ControllerBase.php on line 170
[29-Dec-2017 07:20:03 Europe/Berlin] ReflectionException: Class \OPNsense\Lldp\General does not exist in /usr/local/opnsense/mvc/app/controllers/OPNsense/Base/ApiMutableModelControllerBase.php:103
Stack trace:
#0 /usr/local/opnsense/mvc/app/controllers/OPNsense/Base/ApiMutableModelControllerBase.php(103): ReflectionClass->__construct('\\OPNsense\\Lldp\\...')
#1 /usr/local/opnsense/mvc/app/controllers/OPNsense/Base/ApiMutableModelControllerBase.php(94): OPNsense\Base\ApiMutableModelControllerBase->getModel()
#2 /usr/local/opnsense/mvc/app/controllers/OPNsense/Base/ApiMutableModelControllerBase.php(83): OPNsense\Base\ApiMutableModelControllerBase->getModelNodes()
#3 [internal function]: OPNsense\Base\ApiMutableModelControllerBase->getAction()
#4 [internal function]: Phalcon\Dispatcher->callActionMethod(Object(OPNsense\Lldpd\Api\GeneralController), 'getAction', Array)
#5 [internal function]: Phalcon\Dispatcher->dispatch()
#6 /usr/local/opnsense/www/api.php(26): Phalcon\Mvc\Application->handle()
#7 {main}
```

Tried with stable and latest master :/

**Comment by fichtner** [comment]
need to see the code :)

**Comment by mimugmail** [comment]
https://github.com/mimugmail/plugins/blob/test/net-mgmt/lldpd/src/opnsense/mvc/app/controllers/OPNsense/Lldpd/Api/GeneralController.php

**Comment by mimugmail** [comment]
Hm, it worked now in test branch mit 373 but not with 273. Then I merged test to lldpd branch and now it's still not working, no matter which version I load :(

**Comment by fichtner** [comment]
I'll fix after pull... ready for pull?

**Comment by mimugmail** [comment]
If you are aware of the error, yes :)

**Comment by fichtner** [comment]
No, but it will be found. Thanks :)

**Pull request merged**